src/py-opentimelineio/opentimelineio/adapters/svg.py

```diff
@@ -28,66 +28,133 @@
 
 # otio
 import opentimelineio as otio
+from xml.etree import ElementTree as ET
+from collections import OrderedDict
 
 # python
 import math
 from random import seed
 from random import random
 from collections import namedtuple
 
-COLORS = {
-    'transparent': (0, 0, 0, 0),
-    'black': (0.0, 0.0, 0.0, 255.0),
-    'white': (255.0, 255.0, 255.0, 255.0),
-    'transluscent_white': (255.0, 255.0, 255.0, 178.5),
-    'purple': (127.5, 0.0, 127.5, 255.0),
-    'light_blue': (134.895, 206.04, 235.11, 255.0),
-    'blue': (0.0, 0.0, 255.0, 255.0),
-    'dark_blue': (0.0, 0.0, 137.7, 255.0),
-    'green': (0.0, 127.5, 0.0, 255.0),
-    'dark_green': (0.0, 99.45, 0.0, 255.0),
-    'yellow': (255.0, 255.0, 0.0, 255.0),
-    'gold': (255.0, 214.2, 0.0, 255.0),
-    'orange': (255.0, 164.985, 0.0, 255.0),
-    'red': (255.0, 0.0, 0.0, 255.0),
-    'dark_red': (137.7, 0.0, 0.0, 255.0),
-    'brown': (137.7, 68.85, 25.5, 255.0),
-    'pink': (255.0, 191.25, 201.45, 255.0),
-    'gray': (127.5, 127.5, 127.5, 255.0),
-    'dark_gray': (168.3, 168.3, 168.3, 255.0),
-    'dark_gray_transluscent': (168.3, 168.3, 168.3, 200.0)
-}
+ET.sorted = lambda x: x
+try:
+    ET.Element = ET._Element_Py
+
+    def SubElement(parent, tag, attrib=OrderedDict(), **extra):
+        attrib = attrib.copy()
+        attrib.update(extra)
+        element = parent.makeelement(tag, attrib)
+        parent.append(element)
+        return element
+
+    ET.SubElement = SubElement
+except AttributeError:
+    pass  # nothing else for python2, ElementTree is pure python
```

Comment:
ETree did not preserve attribut le ordering before python3.8.
I didn't go deep into what is actually going on here. But I found a solution on stackoverflow that replacee the dict that ETree uses for attributes with an ordered dict. So the attribute order when we convert the tree to a string is preserved.

Reply:
Hrm, this is quite a bit hackier than I'd prefer - it assumes knowledge about the implementation and usage of `SubElement` beyond what's possible. It would be much better if we could avert dependency on preserving the ordering.

Reply:
I agree coercing behavior (through monkeypatching or anything else) is a "code smell," i.e. a pattern that points to a more basic underlying problem.

I'm trying to follow the attribute ordering conversation, could someone help me out here? Earlier in the thread there's comments that it may or may not be important and/or a problem to preserve attribute ordering. The reason to use ETree is for transmogrification into svg, because ETree is idiomatic python for treating XML, yes? If the idiomatic XML library for Python doesn't map onto SVG naturally, it's probably not the right solution? At the end of the day it's a fairly trivial thing we are representing, so I'm starting to wonder if we're going to spend time coercing the proverbial square peg into the round hole to satisfy idiom, at the expense of just writing the simplest needed code to solve the problem. 

Anyway, if someone could help me understand the issue a bit, I'd really appreciate it.

Reply:
ETree doesn't preserve the order of attributes, when serializing or deserializing XML, before python 3.8. Currently I am just checking string equality in the tests, which fails in python3.8. This is why I did this hacky stuff to replace dict with ordered dict, to force it to maintain attribute order in python versions before 3.8.

As you and Eric mentioned, this approach seems wrong, and I should look for a different way to test.

Reply:
Ah, got it. Looks like ETree was never designed with testing in mind.  This SO thread https://stackoverflow.com/questions/7905380/testing-equivalence-of-xml-etree-elementtree#24349916 is filled with bad ideas and terrible advice!

Reply:
Oh. I was thinking of using etree to parse and check equality of each attribute. That's what the SO thread says right? What would be a better way?

Reply:
Well, the thread says a lot of unhelpful things like "XML is text so use tostring()". My comment that it was not designed with testing in mind comes from the point that testing the nodes for equality directly results in apparent failures even for trivial examples.

Parsing and checking equality is also in the thread, and to me it seems like the only solution, so, yes.

Reply:
And, to close the loop on why use ETree for XML - I recommended that as opposed to using string concatenation to generate the XML because it is important to make sure text from OTIO files is escaped properly. Basically it's the [Bobby Tables](https://xkcd.com/327/) thing.

Reply:
escaping XML is tricky enough that I buy that argument :)